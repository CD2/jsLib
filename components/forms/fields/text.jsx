import React from 'react'
import PropTypes from 'prop-types'

import { isFormField } from '../index'

@isFormField
export default class TextField extends React.Component {

  static propTypes = {
    error: PropTypes.array,
    value: PropTypes.string,
    onChange: PropTypes.onChange,
    type: PropTypes.string,
  }

  static defaultProps = {
    type: 'text',
  }

  handleChange = (e) => {
    this.props.onChange(e.target.value)
  }

  render() {
    const { type, value } = this.props
    return (
      <input type={type} onChange={this.handleChange} value={value}/>
    )
  }

}
