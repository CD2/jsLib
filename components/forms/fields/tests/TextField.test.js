/* eslint-disable react/jsx-filename-extension */
import React from "react"
import { TextField } from "../text"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"

describe(`<TextField />`, () => {
  const view = shallow(<TextField />)

  it(`snapshot`, () => expect(toJson(view)).toMatchSnapshot())
})
