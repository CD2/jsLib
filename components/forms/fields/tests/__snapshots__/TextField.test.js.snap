// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> snapshot 1`] = `
<input
  onChange={[Function]}
  type="text"
  value=""
/>
`;
